test/testdata/policy-controller/e2e/cip-match-resource-label.yaml

```diff
@@ -0,0 +1,36 @@
+# Copyright 2022 The Sigstore Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: policy.sigstore.dev/v1alpha1
+kind: ClusterImagePolicy
+metadata:
+  name: image-policy-match-label
+spec:
+  images:
+  - glob: "**demo**"
+  match:
+  - resource: pods
+    version: v1
```

Comment:
I wonder if we should have a '*' for these, for things like cronjobs? Or maybe I guess you could for cronjobs then have to add both of them (v1, v1beta1?), but then we'd have to support it in the lookup table, which now just has v1 there.
WDYT?

Reply:
I think we can use '*', and add the missing versions to the supported api types. That looks more robust in my opinion. In the future we could add a checking in the policy validating webhook, so the types are not hardcoded in the webhook itself.